pubsub/include/marlin/pubsub/PubSubNode.hpp

```diff
@@ -679,6 +690,53 @@ int PUBSUBNODETYPE::did_recv_MESSAGE(
 			);
 		}
 
+		if(!transport.is_internal() && check_reward_worthy(bytes.data(), bytes.size())) {
+			ABCInterface abci;
+			core::WeakBuffer blockHeader = abci.get_header(bytes);
+			uint8_t hash[32];
+			CryptoPP::Keccak_256 hasher;
+			hasher.CalculateTruncatedDigest(hash, 32, blockHeader.data(), blockHeader.size());
+			
+			uint8_t* key = abci.get_key();
+			if(key == nullptr) {
+				std::cout << "Key not created" << std::endl;
+				return -1;
+			}
+
+			secp256k1_context* ctx_signer = secp256k1_context_create(SECP256K1_CONTEXT_SIGN);
+
+			// Sign
+			secp256k1_ecdsa_recoverable_signature sig;
+			auto res = secp256k1_ecdsa_sign_recoverable(
+				ctx_signer,
+				&sig,
+				hash,
+				key,
+				nullptr,
+				nullptr
+			);
+
+			if(res == 0) {
+				// Sign failed
+				std::cout << "Signing of message failed" << std::endl;
+				return -1;
+			}
+
+			// Output
+			int recid;
+			core::Buffer out(65);
+			secp256k1_ecdsa_recoverable_signature_serialize_compact(
+				ctx_signer,
+				out.data(),
+				&recid,
+				&sig
+			);
+
+			out.data()[64] = (uint8_t)recid;
+
+			transport.send(std::move(out));
```

Comment:
Can't just `transport.send`, it needs its own message type prefixed to indicate its a receipt. Otherwise, your receipt might be interpreted as a SUBSCRIBE instead at the receiver.